contrib/babelfishpg_tsql/src/pltsql_utils.c

```diff
@@ -1574,6 +1576,71 @@ split_object_name(char *name)
 	return res;
 }
 
+/*
+ * Wrapper over split_object_name function above to expose it as a SQL function.
+ */
+Datum
+split_identifier_internal(PG_FUNCTION_ARGS)
+{
+	FuncCallContext	*funcctx;
+	char        	**split_parts = NULL;
+
+	if (SRF_IS_FIRSTCALL())
+	{
+		int         	num_parts = 0;
+		MemoryContext	oldcontext;
+
+		funcctx = SRF_FIRSTCALL_INIT();
+		oldcontext = MemoryContextSwitchTo(funcctx->multi_call_memory_ctx);
+		if (!PG_ARGISNULL(0))
+		{
+			char	*input;
+			char	**splited_object_name;
+			int 	i, j = 0;
+
+			input = text_to_cstring(PG_GETARG_TEXT_P(0));
+			splited_object_name = split_object_name(input);
+
+			for (i = 0; i < 4; i++)
+			{
+				if (strlen(splited_object_name[i]) > 0)
+					num_parts++;
+			}
+
+			if (num_parts > 0)
+			{
+				split_parts = (char **) palloc(num_parts * sizeof(char *));
+
+				for (i = 0; i < 4; i++)
+				{
+					if (i >= (4 - num_parts))
+						split_parts[j++] = splited_object_name[i];
+					else
+						pfree(splited_object_name[i]);
+				}
+			}
+			pfree(splited_object_name);
+		}
+
+		funcctx->max_calls = num_parts;
+		funcctx->user_fctx = split_parts;
+		MemoryContextSwitchTo(oldcontext);
+	}
+
+	funcctx = SRF_PERCALL_SETUP();
+	split_parts = (char **) funcctx->user_fctx;
+
+	if (funcctx->call_cntr < funcctx->max_calls)
+	{
+		VarChar *val = (*common_utility_plugin_ptr->tsql_varchar_input) (split_parts[funcctx->call_cntr],
+																		 strlen(split_parts[funcctx->call_cntr]),
+																		 -1);
```

Comment:
Is it ok not to free split_parts[funcctx->call_cntr] after calling common_utility_plugin_ptr->tsql_varchar_input ?